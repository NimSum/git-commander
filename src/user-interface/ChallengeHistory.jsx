import React from 'react';

export function ChallengeHistory(props) {
  return (
    <section className="challenge-history">
      <h3>Challenge History:</h3>
      {props.challenges.length 
        ? props.challenges.map((challenge, i) => {
        return (
          <article key={i}>
            <p class="challenge">Challenge: <span>{challenge.detail}</span></p>
            <p class="command">Command: <span>{challenge.answer}</span></p>
            <p><a
              target="_blank"
              rel="noopener noreferrer" 
              href="https://gist.github.com/NimSum/f190231f92183f4851074ef2502537c6">
              Learn More</a>
            </p>
          </article>
        )})
      : <h4>None Yet</h4> }
    </section>)
}

export default ChallengeHistory;