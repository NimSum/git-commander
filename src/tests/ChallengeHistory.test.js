import React from 'react';
import ChallengeHistory from '../user-interface/ChallengeHistory';
import Adapter from 'enzyme-adapter-react-16';
import { shallow , configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Game Window' , () => {
  
  let wrapper;
  const mockQuestion =  {
    "statement" : "Our Git spaceship needs to switch to a different branch to steer the ship!",
    "detail" : "Switch to the branch called \"steer-ship\" to avoid the obstacle",
    "answer" : "git checkout steer-ship", 
    "difficulty" : "1",
    "moreInfo" : "/////A link to where they could read more about this command",
    "id": "1"
  };
  
  beforeEach(() => {
    wrapper = shallow(
      < ChallengeHistory 
        challenges={ mockQuestion }
      />
    );
  });

  it('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
});