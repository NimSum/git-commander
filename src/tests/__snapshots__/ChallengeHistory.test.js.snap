// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Window Should match snapshot 1`] = `
<section
  className="challenge-history"
>
  <h3>
    Challenge History:
  </h3>
  <h4>
    None Yet
  </h4>
</section>
`;
